tests/integration/synapseutils/test_synapseutils_sync.py

```diff
@@ -294,11 +294,24 @@ def test_syncFromSynapse(test_state):
         uploaded_paths.append(f)
         test_state.schedule_for_cleanup(f)
         test_state.syn.store(File(f, parent=folder_entity))
+
     # Add a file in the project level as well
     f = utils.make_bogus_data_file()
     uploaded_paths.append(f)
     test_state.schedule_for_cleanup(f)
-    test_state.syn.store(File(f, parent=project_entity))
```

Comment:
@jaymedina if you look in this class, you'll see `schedule_for_cleanup` functions, but it doesn't setup / teardown per test.

Reply:
I see - It might be good to tear down per test because I turned off pytest `reruns` for this particular integration test, which uncovered the real error that was breaking this test. Or maybe consider turning off `reruns` entirely for integration tests where the Synapse data repositories get updated.

```
>           raise SynapseHTTPError(message, response=response)
E           synapseclient.core.exceptions.SynapseHTTPError: 404 Client Error: 
E           Cannot find a node with id syn53404745 and version 2
```

Reply:
> It might be good to tear down per test

If you have a suggestion to add feel free - Perhaps we could spy on the client and keep track of all of the things that need to be cleaned up and add a finalizer automatically.

Ideally everything created in an integration test is deleted.


Some background on the `reruns` - I added this in a few months ago as a band-aid fix. It got us into a "good enough" position to not spend the time to look further at the individual failures and patch each thing that randomly fails for us.



Also - As a heads up if you are running these integration tests make sure you are pointing to the dev endpoints (If you aren't already):
```
[endpoints]
repoEndpoint=https://repo-dev.dev.sagebase.org/repo/v1
authEndpoint=https://repo-dev.dev.sagebase.org/auth/v1
fileHandleEndpoint=https://repo-dev.dev.sagebase.org/file/v1
```

If you need help setting this up reach out.

Reply:
Hi @BryanFauble , the `reruns` background makes sense, thanks for the context. In case that, it makes sense to leave it in for now. I think we would benefit more from cleanups after each test execution (pass or fail). That way each rerun starts on a blank canvas.

Reply:
Agreeded

Reply:
I've made a Jira ticket for this: https://sagebionetworks.jira.com/browse/SYNPY-1424